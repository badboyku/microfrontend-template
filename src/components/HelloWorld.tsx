import * as React from 'react';

type Props = {};

const HelloWorld = (_props: Props): JSX.Element => {
  return <div>Hello World!</div>;
};

export default HelloWorld;
