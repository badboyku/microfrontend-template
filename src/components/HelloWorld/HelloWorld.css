.hello-world {
  color: #7d61fb;
}

.logo {
  width: 40px;
}
