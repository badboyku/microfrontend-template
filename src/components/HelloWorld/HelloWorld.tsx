import * as React from 'react';
import logoUrl, {ReactComponent as Logo} from './logo.svg';
import './styles.css';

type Props = {};

const HelloWorld = (_props: Props): JSX.Element => {
  return (
    <div>
      <div className="hello-world">Hello World!!!</div>
      <img src={logoUrl} className="logo" alt="logo" />
      <Logo width={40} />
    </div>
  );
};

export default HelloWorld;
