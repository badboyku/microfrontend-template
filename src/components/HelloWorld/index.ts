/* istanbul ignore file */
import HelloWorld from './HelloWorld';

export default HelloWorld;
