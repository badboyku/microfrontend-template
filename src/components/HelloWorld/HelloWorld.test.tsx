import * as React from 'react';
import {render} from '@testing-library/react';
import HelloWorld from './HelloWorld';

const renderHelloWorld = (props = {}) => {
  return render(<HelloWorld {...props} />);
};

describe('Component HelloWorld', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderHelloWorld();
    expect(asFragment()).toMatchSnapshot();
  });
});
