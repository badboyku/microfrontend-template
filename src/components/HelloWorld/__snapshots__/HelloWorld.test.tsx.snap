// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HelloWorld renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="hello-world"
    >
      Hello World!!!
    </div>
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <svg
      width="40"
    >
      logo.svg
    </svg>
  </div>
</DocumentFragment>
`;
